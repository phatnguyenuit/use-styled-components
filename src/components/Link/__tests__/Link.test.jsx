import { Link, StyledLink } from "../index";

import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("Test `Links` with `react-test-renderer`", () => {
  test("render `Link` without error", () => {
    const link = renderer.create(<Link>Link Text</Link>);
    let tree = link.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("render `StyledLink` without error", () => {
    const link = renderer.create(<StyledLink>Styled Link</StyledLink>);
    let tree = link.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("Test `Links` with `enzyme`", () => {
  test("render `Link` with text", () => {
    const link = shallow(<Link>Link Text</Link>);
    expect(link.contains("Link Text")).toEqual(true);
  });
  test("render `StyledLink` with text", () => {
    const link = shallow(<StyledLink>Styled Link</StyledLink>);
    expect(link.contains("Styled Link")).toEqual(true);
  });
});
