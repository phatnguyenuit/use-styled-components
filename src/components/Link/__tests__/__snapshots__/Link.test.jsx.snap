// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test \`Links\` with \`react-test-renderer\` render \`Link\` without error 1`] = `
<a
  className="sc-bdVaJa bpBYMg"
>
  Link Text
</a>
`;

exports[`Test \`Links\` with \`react-test-renderer\` render \`StyledLink\` without error 1`] = `
<a
  className="sc-bdVaJa sc-bwzfXH dsehKE"
>
  Styled Link
</a>
`;
